internal/cltest/mocks.go

```diff
@@ -632,3 +634,16 @@ func (m *MockRunChannel) Receive() <-chan store.RunRequest {
 }
 
 func (m *MockRunChannel) Close() {}
+
+// ExtractTargetAddressFromERC20EthEthCallMock extracts the contract address and the
+// method data, for checking in a test.
+func ExtractTargetAddressFromERC20EthEthCallMock(
+	t *testing.T, arg ...interface{}) common.Address {
+	args, ok := (arg[0]).([]interface{})
+	require.True(t, ok)
+	args, ok = (args[0]).([]interface{})
```

Comment:
Is this supposed to be `args[1]`?

Reply:
You'd think so, but here is the argument to `ExtractTargetAddressFromERC20EthEthCallMock`:

```
[[[{[81 73 16 119 26 249 202 101 106 248 64 223 248 62 130 100 236 249 134 202] 0x70a082310000000000000000000000003141592653589793238462643383279502884197} latest]]]
```

The internal struct is a `store.callArgs`, which `store.ExtractERC20BalanceTargetAddress` gets the address from. I haven't investigated why it's so deeply nested.

Reply:
Ok, but then why are we making the same assertion twice?

```golang
args, ok := (arg[0]).([]interface{})
require.True(t, ok)
args, ok := (arg[0]).([]interface{})
require.True(t, ok)
```

Am I missing something?

Reply:
You reassign `args`, got it.

Reply:
Yep, that's it. Shows it's a bit confusing, though. Will fix.